index.html

```diff
@@ -1296,6 +1296,53 @@ <h3>Types</h3>
 }
         </pre>
 
+        <p>
+With respect to this specification, there are at least three different
+categories of type information that implementers are advised to be aware
+of when building solutions. These are:
+        </p>
+
+        <ol>
+          <li>
+The type of the <a>verifiable credential</a>. This signals the top level
+properties that the verifiable credential contains to the <a>verifier</a>.
+The type <code>VerifiableCredential</code> signals all of the properties that
+a basic <a>verifiable credential</a> contains. These properties are all defined
+in this specification and include <code>credentialSubject</code>,
+<code>issuer</code>, <code>proof</code>, and others. If an <a>issuer</a> wants
+to add new top level properties to a <a>verifiable credential</a>, then it
+is strongly advised that the type property be extended with an additional type
+value that indicates that these new properties exist. It is also strongly
+advised that the <code>credentialSchema</code> be utilized to define which of
+these new properties are mandatory and which are optional.
+          </li>
+          <li>
+The type of the credential <a>subject</a>. Since the
+<code>credentialSubject</code> is simply identified by an identifier, the
+type of the <a>subject</a> might not be obvious to a <a>verifier</a>.
+For example, a temperature sensor might be one that reads the temperature of an
+oven, a person, or a refrigerator. Specifying identifier and temperature
+properties only might not allow the verifier to determine whether the sensor
+is operating correctly or not. Specifying the type of sensor will allow the
+<a>verifier</a> to know the expected range of temperature values.
+An <a>issuer</a> is strongly advised to express the type of
+the credential <a>subject</a> unless it might lead to issues such as privacy
+violations of the <a>subject</a>.
+          </li>
+          <li>
+The type that governs the properties of the credential <a>subject</a>.
+For example, a <a>verifiable credential</a> of type <code>UniversityDegree</code>
+might contain the discipline and degree classification, whilst an
+<code>ExtendedUniversityDegree</code> might contain a list of courses studied
+for the degree. An issuer might specify this additional subtype information in
+<code>type</code> property of the <a>verifiable credential</a> or might leave
+the subtype to be implicitly determined by the verifier. For example,
+when the <a>verifier</a> finds additional unexpected properties associated with
+the credential <a>subject</a> in the <a>verifiable credential</a> then it knows
+that the type has been extended.
+          </li>
```

Comment:
```pre
The type that governs the properties of the credential <a>subject</a>.
For example, a <a>verifiable credential</a> of type <code>UniversityDegree</code>
might contain the discipline and degree classification, whilst an
<code>ExtendedUniversityDegree</code> might contain a list of courses studied
for the degree. An issuer might specify this additional subtype information in
<code>type</code> property of the <a>verifiable credential</a> or might leave
the subtype to be implicitly determined by the verifier. For example,
when the <a>verifier</a> finds additional unexpected properties associated with
the credential <a>subject</a> in the <a>verifiable credential</a> then it knows
that the type has been extended.
```
->
```pre
             <p>
The subtype of <a>verifiable credential</a> (communicated as the value of a
<a>type</a> property), which describes which properties of the credential
<a>subject</a> are expected to be found in a <a>verifiable credential</a> of
that subtype. Such properties might be described as mandatory or optional. For
example, a <a>verifiable credential</a> of subtype <code>UniversityDegree</code>
might be described to contain mandatory properties including the discipline and
degree classification, while a <a>verifiable credential</a> of subtype
<code>ExtendedUniversityDegree</code> might optionally contain a list of courses
studied for the degree.
             </p>
             <p>
An issuer might specify this additional subtype information in the <code>type</code>
property of the <a>verifiable credential</a>, or might leave the subtype to be
implicitly determined (or "inferred") by the verifier. For example, when a
<a>verifier</a> finds additional unexpected properties associated with the
credential <a>subject</a> in the <a>verifiable credential</a>, then the verifier
knows that the base type has been extended.  Verifiers might set policies that
refuse all action if a <a>verifiable credential</a> contains properties not
enumerated by its explicitly declared <a>type</a>; or set policies that refuse
any action based on a property not enumerated by an explicitly declared <a>type</a>;
simply ignore such extra properties; or treat the extra properties as just as
authoritative as anything else in the <a>verifiable credential</a>, based on
their treatment of the issuer as universally authoritative. There are other
possibilities.
```

That's pretty extensive.  It probably wants more text, pulling from the extensive discussion threads on this topic.  Or this whole PR could be omitted entirely.